packages/next-server/server/next-server.ts

```diff
@@ -371,11 +384,28 @@ export default class Server {
     }))
   }
 
+  private handleCompression(
+    req: IncomingMessage,
+    res: ServerResponse
+  ): Promise<void> {
+    return new Promise((resolve, reject) => {
+      if (!this.compression) {
+        return resolve()
+      }
+      this.compression(req, res, err => {
```

Comment:
It's not possible for the latter function to be invoked with an error by this specific handler, as it's just invoking `next()`.

Would it be better to drop all the async nature of this and just move the logic down below?